src/components/CameraControls/index.tsx

```diff
@@ -1,26 +1,152 @@
-import * as React from "react";
+import React, { useEffect, useState, useRef } from "react";
 import { Button, Tooltip } from "antd";
+import classNames from "classnames";
+import { useHotkeys } from "react-hotkeys-hook";
 
 import { TOOLTIP_COLOR } from "../../constants/index";
 import Icons from "../Icons";
 
 const styles = require("./style.css");
 
+const PAN = "pan";
+const ROTATE = "rotate";
+const CAMERA_MODE_MODIFIER_KEYS = ["Meta", "Shift"];
+const ZOOM_IN_HK = "ArrowUp";
+const ZOOM_OUT_HK = "ArrowDown";
+const RESET_HK = "r";
+const HOT_KEYS = [ZOOM_IN_HK, ZOOM_OUT_HK, RESET_HK];
+
 interface CameraControlsProps {
     resetCamera: () => void;
     zoomIn: () => void;
     zoomOut: () => void;
+    setPanningMode: (value: boolean) => void;
 }
 
 const CameraControls = ({
     resetCamera,
     zoomIn,
     zoomOut,
+    setPanningMode,
 }: CameraControlsProps) => {
+    const [mode, setMode] = useState(ROTATE);
+    const [keyPressed, setKeyPressed] = useState("");
+    const lastKeyPressed = useRef("");
+    const isModifierKey = (key: string) =>
+        CAMERA_MODE_MODIFIER_KEYS.includes(key);
+    useHotkeys(
+        "*",
+        (event) => {
+            if (
+                CAMERA_MODE_MODIFIER_KEYS.includes(event.key) ||
+                HOT_KEYS.includes(event.key)
+            ) {
+                return setKeyPressed(event.key);
+            }
+        },
+        { keydown: true }
+    );
+
+    useHotkeys(
+        "*",
+        () => {
+            return setKeyPressed("");
+        },
+        { keyup: true }
+    );
+
+    useEffect(() => {
+        if (
+            !isModifierKey(lastKeyPressed.current) &&
+            !isModifierKey(keyPressed)
+        ) {
+            // only call the simularium controller if it's not a modifier key
+            // because the viewer is already setting the mode
+            setPanningMode(mode === PAN);
+        }
+    }, [mode]);
+
+    useEffect(() => {
+        if (
+            (isModifierKey(keyPressed) && !lastKeyPressed.current) ||
+            (isModifierKey(lastKeyPressed.current) && !keyPressed)
+        ) {
+            // toggle the mode, so the radio buttons reflect the true state
+            lastKeyPressed.current = keyPressed;
+            return setMode(mode === PAN ? ROTATE : PAN);
+        }
+        switch (keyPressed) {
+            case ZOOM_IN_HK:
+                zoomIn();
+                break;
+            case ZOOM_OUT_HK:
+                zoomOut();
+                break;
+            case RESET_HK:
+                resetCamera();
+                break;
+            default:
+                break;
+        }
+        lastKeyPressed.current = keyPressed;
+    }, [keyPressed]);
+
     return (
         <div className={styles.container}>
+            <div className={styles.moveButtons}>
```

Comment:
Is this div necessary? (I don't see a `move-buttons` selector in the CSS)

Reply:
I use it in the next PR to group the move buttons together